src/Solvers/col_projection.jl

```diff
@@ -0,0 +1,326 @@
+"""
+    col_projection <: Solver
+
+An implementation of a column projection solver. Specifically, it is a solver that iteratively
+    updates a solution by projection the solution onto a compressed rowspace of the linear 
+    system.
+
+# Mathmatical Description 
+
+Let `A` be an `m \\times n` matrix and consider the consistent linear system `Ax = b`. 
```

Comment:
This does not have to be for a consistent linear system. Any least squares problem works.